### PR TITLE
feat: adds platform config to task definition to support ARM

### LINKED ISSUE #1891: Unable to configure runtimePlatform in ECS Task Definition to use ARM
### Summary

I am trying to create task definitions via Ansible, however runtimePlatform can not be configured, this means it is always the default:
```
    "runtimePlatform": {
        "cpuArchitecture": "X86_64",
        "operatingSystemFamily": "LINUX"
    },
```

Runtime Platform should be configurable to make use of ARM instances on AWS Fargate. And I am pretty sure, this way also Windows systems on fargate could be supported, however no idea if fargate supports windows currently, windows is not my area :-D

Currently I am writing a patch to support this, the issue is just to link my PR :)

### Issue Type

Feature Idea

### Component Name

ecs_task_definition

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by romulus-ai**
##### SUMMARY
Fixes #1891 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ecs_taskdefinition

##### ADDITIONAL INFORMATION
I've just added the Parameter the same way the other parameters are also included. Tested it with our AWS Account.

### CODE REVIEW

**Comment by romulus-ai** [comment]
The PR is against main, or should I make it against stable-6? Howver, if I do so, I get merge conflicts.

**Comment by tremble** [comment]
Thanks for taking the time to open this PR, it's very much appreciated.

I've not got the cycles to review right now, hopefully someone else will have some time.

However, I wanted to quickly answer you question:

> The PR is against main, or should I make it against stable-6? Howver, if I do so, I get merge conflicts.

Generally PRs should be against "main", we then backport from main to stable-6.


It will also help if you can add some integration tests (tests/integration/targets/ecs_cluster) they're just an Ansible role.


Mark

**Comment by romulus-ai** [comment]
@tremble Added some basic tests, you're right, better tests better code ;)

**Comment by romulus-ai** [comment]
@Java1Guy @alinabuzachis @jillr @markuman @s-hertel @tremble so far I fixed everything complained by CI, however, I have no Idea what Tox is complaining here, do one of you may know?

"would reformat /home/runner/work/community.aws/community.aws/plugins/modules/ecs_taskdefinition.py" is the only information that I see, however there is nothing in detail what's disturbing him. Would be nice if someone could help :)

**Comment by alinabuzachis** [comment]
> @Java1Guy @alinabuzachis @jillr @markuman @s-hertel @tremble so far I fixed everything complained by CI, however, I have no Idea what Tox is complaining here, do one of you may know?
> 
> "would reformat /home/runner/work/community.aws/community.aws/plugins/modules/ecs_taskdefinition.py" is the only information that I see, however there is nothing in detail what's disturbing him. Would be nice if someone could help :)

@romulus-ai  Please reformat using black the `plugins/modules/ecs_taskdefinition.py` files. This collection uses black formatting. FYI https://github.com/ansible-collections/community.aws/blob/main/tox.ini#L31

File: plugins/modules/ecs_taskdefinition.py [hunk 1]
```diff
@@ -627,6 +627,26 @@
             expression:
                 description: A cluster query language expression to apply to the constraint.
                 type: str
+    runtime_platform:
+        description:
+            - runtime platform configuration for the task
+        required: false
```
Comment (alinabuzachis) [inline plugins/modules/ecs_taskdefinition.py:633]: Can you please add `version_added: 6.2.0` ?

File: plugins/modules/ecs_taskdefinition.py [hunk 2]
```diff
@@ -627,6 +627,26 @@
             expression:
                 description: A cluster query language expression to apply to the constraint.
                 type: str
+    runtime_platform:
+        description:
+            - runtime platform configuration for the task
```
Comment (alinabuzachis) [inline plugins/modules/ecs_taskdefinition.py:632]:
```suggestion
            - Runtime platform configuration for the task.
```

File: plugins/modules/ecs_taskdefinition.py [hunk 3]
```diff
@@ -627,6 +627,26 @@
...
+        required: false
+        type: dict
+        default: {
+                    "operatingSystemFamily": "LINUX",
+                    "cpuArchitecture": "X86_64"
+                }
+        suboptions:
+            cpuArchitecture:
+                description: The CPU Architecture type to be used by the task
```
Comment (alinabuzachis) [inline plugins/modules/ecs_taskdefinition.py:641]:
```suggestion
                description: The CPU Architecture type to be used by the task.
```

File: plugins/modules/ecs_taskdefinition.py [hunk 4]
```diff
@@ -627,6 +627,26 @@
...
+                type: str
+                required: false
+                choices: ['X86_64', 'ARM64']
+            operatingSystemFamily:
+                description: OS type to be used by the task
```
Comment (alinabuzachis) [inline plugins/modules/ecs_taskdefinition.py:646]:
```suggestion
                description: OS type to be used by the task.
```

File: plugins/modules/ecs_taskdefinition.py (hunk 1 above)
Comment (markuman) [reply plugins/modules/ecs_taskdefinition.py:633]: @romulus-ai

Comment (romulus-ai) [reply plugins/modules/ecs_taskdefinition.py:634]: done, delay was because of holiday

**Comment by romulus-ai** [comment]
No Idea what going wrong now, anyone else?

**Comment by markuman** [comment]
recheck

**Comment by markuman** [comment]
@romulus-ai  integration test is failing in the task `TASK [ecs_cluster : create task definition for ARM] `


>  "msg": "Failed to register task: Parameter validation failed:\nInvalid type for parameter runtimePlatform.operatingSystemFamily, value: None, type: <class 'NoneType'>, valid types: <class 'str'>",

**Comment by romulus-ai** [comment]
@markuman fixed them, however sanity is still red, any Idea?

**Comment by markuman** [comment]
> @markuman fixed them, however sanity is still red, any Idea?

hmmmm

> ERROR: tests/unit/mock/loader.py:33:16: E721: do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`

I am not sure....

cc @alinabuzachis @tremble

**Comment by romulus-ai** [comment]
@alinabuzachis @tremble any Idea? Would be nice to have this finished soon :-)

**Comment by markuman** [comment]
recheck

**Comment by markuman** [comment]
@romulus-ai  can you rebase your branch with the current main branch please?

**Comment by romulus-ai** [comment]
@markuman done

File: plugins/modules/ecs_taskdefinition.py [hunk 5]
```diff
@@ -627,6 +627,27 @@
             expression:
                 description: A cluster query language expression to apply to the constraint.
                 type: str
+    runtime_platform:
+        version_added: 6.2.0
```
Comment (markuman) [inline plugins/modules/ecs_taskdefinition.py:631]:
```suggestion
        version_added: 6.4.0
```

**Review by markuman — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**